File: backend/app/core/schemas/news.py
from pydantic import BaseModel
from datetime import date


class SentimentCount(BaseModel):
    name: str
    count: int


class SentimentByDay(BaseModel):
    date: date
    positive: int
    negative: int
    neutral: int
    very_negative: int
    very_positive: int
    all: int


class SentimentByCountry(BaseModel):
    country: str
    positive: int
    negative: int
    neutral: int
    very_negative: int
    very_positive: int
    all: int

File: src/lib/types/news.ts
// Type for daily sentiment data
export interface sentimentStatsType {
  date: string;
  positive: number;
  negative: number;
  neutral: number;
  very_negative: number;
  very_positive: number;
}
[];
export interface sentimentDistributionType {
  name: string;
  value: number;
  color: string;
}
[];

export interface topCountriesType {
  id: number;
  country: string;
  count: number;
}

export interface countrySentimentType {
  id: number;
  country: string;
  positive: number;
  neutral: number;
  negative: number;
  very_positive: number;
  very_negative: number;
}
